datadotworld/client/api.py

```diff
@@ -0,0 +1,451 @@
+"""
+from __future__ import absolute_import, division
+
+import json
+import os
+import shutil
+import uuid
+import zipfile
+from os import path
+
+import requests
+
+from datadotworld.client import _swagger
+from datadotworld.config import Config
+from datadotworld.util import parse_dataset_key, _user_agent
+
+
+class RestApiClient(object):
+    """A Python Client for data.world's REST API
+
+    Parameters
+    ----------
+    profile : str, optional
+        Name of the configuration profile to use
+    """
+    def __init__(self, config):
+        self._config = config
+        self._protocol = 'https'
+        self._download_host = 'download.data.world'
+
+        api_host = 'api.data.world'
+        swagger_client = _swagger.ApiClient(host="{}://{}/v0".format(self._protocol, api_host),
+                                            header_name='Authorization',
+                                            header_value='Bearer {}'.format(self._config.auth_token))
+        swagger_client.user_agent = _user_agent()
+
+        self._datasets_api = _swagger.DatasetsApi(swagger_client)
+        self._uploads_api = _swagger.UploadsApi(swagger_client)
+
+    # Dataset Operations
+
+    def get_dataset(self, dataset_key):
+        """Retrieve an existing dataset
+
+        Parameters
+        ----------
+        dataset_key : str
+            Dataset identifier, in the form of owner/id
+
+        Returns
+        -------
+        dict
+            Dataset definition, with all attributes
+
+        Raises
+        ------
+        RestApiException
+            If a server error occurs
+
+        Examples
+        --------
+        >>> api_client = datadotworld.api_client()
+        >>> intro_dataset = api_client.get_dataset('jonloyens/an-intro-to-dataworld-dataset')
+        >>> print(intro_dataset['description'])
+        A dataset that serves as a quick introduction to data.world and some of our capabilities.  Follow along in \
+        the summary!
+        """
+        try:
+            return self._datasets_api.get_dataset(*(parse_dataset_key(dataset_key))).to_dict()
+        except _swagger.ApiException as e:
+            raise RestApiError(cause=e)
+
+    def create_dataset(self, owner_id, **kwargs):
+        """Create a new dataset
+
+        Parameters
+        ----------
+        owner_id : str
+            Username of the owner of the new dataset
+        title : str
+            Dataset title
+        description : str, optional
+            Dataset description
+        summary : str, optional
+            Dataset summary markdown
+        tags : list, optional
+            Dataset tags
+        license : {'Public Domain', 'PDDL', 'CC-0', 'CC-BY', 'ODC-BY', 'CC-BY-SA', 'ODC-ODbL', 'CC BY-NC', 
+                   'CC BY-NC-SA', 'Other'}
+            Dataset license
+        visibility : {'OPEN', 'PRIVATE'}
+            Dataset visibility
+        files : dict
```

Comment:
What are the keys and values of the dict?  Is this `names -> urls`?  Should be optional, and there should be text explaining that the `files` dict is only relevant if the files-to-add are publicly available on the internet, direct upload is possible via a different api call.